content/blog/gmailr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,183 @@
+date: 2023-06-29
+author: Jennifer Bryan
+description: >
+    gmailr 2.0.0 streamlines the auth process and makes it easier to use gmailr
+    in a cloud or deployed context.
+
+photo:
+  url: https://unsplash.com/photos/VybzKEUMhbw
+  author: Hiroshi Kimura
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package] 
+tags: [gmailr, gargle]
+---
+
+We're chuffed to announce the release of [gmailr](https://gmailr.r-lib.org/) 2.0.0.
+gmailr exposes the [Gmail API](https://developers.google.com/gmail/api/guides) from R.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("gmailr")
+```
+
+The main goal of version 2.0.0 is to improve the ergonomics around auth.
+There is less need for fussy code around configuring an OAuth client and it's easier to use gmailr in a non-interactive or deployed setting.
+There is also a major advance in the process of replacing legacy functions with versions that have a `gm_` prefix.
+The legacy functions still exist, but are now hard deprecated.
+Finally, gmailr no longer re-exports `%>%`, the magrittr pipe, now that we have `|>` in base R.
+
+You can see a full list of changes in the [release notes](https://github.com/r-lib/gmailr/releases/tag/v2.0.0).
+
+```{r setup}
+library(gmailr)
+```
+
+😬 *Ouch! These name collisions are **exactly** why gmailr added a universal `gm_` prefix to all of its functions starting in v1.0.0. One day in the not-too-distant future we can remove the troublesome legacy functions.*
+
+## OAuth client
+
+The Gmail API is more challenging to wrap, in terms of auth, than the APIs for Sheets, Drive, or BigQuery.
+That's because the scopes (think: "permissions") needed for the Gmail API are regarded as extremely sensitive, as well they should be.
+If a bad actor gains the ability to read and send email as you, that is considerably more damaging than them being able to modify your spreadsheets (which is also bad, to be sure, but considerably less bad).
+
+The heightened security around the Gmail API means that a wrapper package like gmailr can't make auth "just work" as easily we can in other packages, such as googledrive.
+In particular, R users who want to use gmailr absolutely must provide their own *OAuth client*.
+In other packages, we can make this optional.
+
+gmailr v2.0.0 includes new features and documentation to reduce the pain around the OAuth client as much as possible:
+
+* [Set up an OAuth client](https://gmailr.r-lib.org/articles/oauth-client.html)
+  is a new article with detailed instructions for creating and
+  configuring an OAuth client. You might even say this provides an excruciating
+  level of detail, but this process has proven to be tricky for many users.
+* There is now a default location for the JSON file that represents the OAuth
+  client. It's the location returned by `rappdirs::user_data_dir("gmailr")`. If
+  you put the JSON file there, gmailr will find it automagically.
```

Comment:
I assume the file name doesn't matter?

Reply:
Well, there is regular expression that checks that the filename "looks" like the ones you download from the developer console. That detail is well documented elsewhere, so I don't think it belongs in the blog post. And it's what will happen by default, i.e. as long as someone doesn't go out of their way to change the name.